### PR TITLE
Telegram

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- adds a ``TelegramInput`` component

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by tmbo** [comment]
Looking good. Can you please merge the current master branch into this feature branch and resolve the conflicts?

**Comment by akelad** [comment]
done!

File: rasa_core/run.py [hunk 1]
```diff
@@ -106,17 +109,22 @@ def _create_external_channel(channel, port, credentials_file):
...
+    elif channel == "telegram":
+        input_blueprint = TelegramInput(
+            credentials.get("access_token"),
+            credentials.get("verify"),
+            credentials.get("webhook_url"))
     else:
-        Exception("This script currently only supports the facebook "
-                  "and slack connectors.")
+        Exception("This script currently only supports the facebook,"
+                  ", telegram and slack connectors.")
```
Comment (ricwo) [inline rasa_core/run.py]: one comma

File: rasa_core/run.py [hunk 2]
```diff
@@ -106,17 +109,22 @@ def _create_external_channel(channel, port, credentials_file):
...
 
     return HttpInputChannel(port, None, input_blueprint)
 
 
 def create_input_channel(channel, port, credentials_file):
     """Instantiate the chosen input channel."""
 
-    if channel == "facebook" or channel == "slack":
+    if channel == "facebook" or "slack" or "telegram":
```
Comment (ricwo) [inline rasa_core/run.py]: this will always be true. how about

`if channel in ['facebook', 'slack', 'telegram']:`

Comment (tmbo) [reply rasa_core/run.py]: I think I already fixed that once for slack? why is it back in here?

Comment (ricwo) [reply rasa_core/run.py]: maybe this was branched from the slack branch before you spotted that

Comment (akelad) [reply rasa_core/run.py]: you did not, but I will fix it :D

Comment (tmbo) [reply rasa_core/run.py]: mhm I did, it is fixed on master :D https://github.com/RasaHQ/rasa_core/blob/master/rasa_core/run.py#L119

Comment (tmbo) [reply rasa_core/run.py]: but yeah, good spot :+1:

Comment (akelad) [reply rasa_core/run.py]: oh yeah sorry, I think I must've deleted the wrong one when resolving conflicts

**Comment by tmbo** [comment]
@akelad this is ready to be merged, right?

**Comment by akelad** [comment]
yep!

**Comment by tmbo** [comment]
Nice 🚀

**Comment by rojobuffalo** [comment]
What is the reasoning behind this syntax:
```
input_channel = TelegramInput(
  access_token="YOUR_ACCESS_TOKEN", # you get this when setting up a bot
  verify="YOUR_BOT_NAME",
  webhook_url="yoursite.com/app/webhook"
)
```
as opposed to passing a `telegram-credentials.yml` file? I like the separate credentials, which has an option on the CLI. But if I'm running a python script, how can I maintain using the separate credentials file?

**Comment by akelad** [comment]
like this https://core.rasa.com/connectors.html#using-run-script

**Comment by akelad** [comment]
You could load the yaml file with `read_yaml_file` from `rasa_nlu.utils` and then get the values you need with `.get()`

**Comment by rojobuffalo** [comment]
Thank you, that's exactly what I looking for.

**Comment by rojobuffalo** [comment]
Think you meant `rasa_core.utils`

I ended up with this:
```
from rasa_core.utils import read_yaml_file

credentials_file = 'telegram_credentials.yml'
credentials = read_yaml_file(credentials_file)
input_channel = TelegramInput(
  access_token=credentials.get("access_token"),
  verify=credentials.get("verify"),
  webhook_url=credentials.get("webhook_url")
)
```

**Pull request merged**